### PR TITLE
Added gtg endpoint

### PR DESCRIPTION
**Pull request by davidbalazs93**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 FROM alpine:3.5
```
Comment (MihaiSorinMoisa) [inline Dockerfile]: We should change to the alpine version image that comes with Go.

Comment (MihaiSorinMoisa) [inline Dockerfile]: We should change to the alpine version image that comes with Go.

File: circle.yml [hunk 2]
```diff
@@ -1,7 +1,23 @@
 machine:
```
Comment (MihaiSorinMoisa) [inline circle.yml]: Coveralls integration should be added.

File: kafka-bridge.go [hunk 3]
```diff
@@ -75,14 +77,19 @@ func initBridgeApp() *BridgeApp {
...
 }
 
-func (bridgeApp *BridgeApp) enableHealthchecks() {
-	//create healthcheck service according to the producer type
+func (bridgeApp *BridgeApp) enableHealthchecksAndGTG() {
+	var gtgHandler func(http.ResponseWriter, *http.Request)
+	//create healthcheck and gtg endpoints according to the producer type
 	if bridgeApp.producerType == proxy {
 		http.HandleFunc("/__health", fthealth.Handler("Dependent services healthcheck", "Services: kafka-rest-proxy@ucs, kafka-rest-proxy@aws", bridgeApp.consumeHealthcheck(), bridgeApp.proxyForwarderHealthcheck()))
+		gtgHandler = httphandlers.NewGoodToGoHandler(gtg.StatusChecker(bridgeApp.proxyGtgCheck))
```
Comment (MihaiSorinMoisa) [inline kafka-bridge.go]: It's not necessary to cast proxyGtgCheck to StatusChecker.

File: kafka-bridge.go [hunk 4]
```diff
@@ -75,14 +77,19 @@ func initBridgeApp() *BridgeApp {
...
 	} else if bridgeApp.producerType == plainHTTP {
 		http.HandleFunc("/__health", fthealth.Handler("Dependent services healthcheck", "Services: kafka-rest-proxy@ucs, cms-notifier@aws", bridgeApp.consumeHealthcheck(), bridgeApp.httpForwarderHealthcheck()))
+		gtgHandler = httphandlers.NewGoodToGoHandler(gtg.StatusChecker(bridgeApp.httpGtgCheck))
```
Comment (MihaiSorinMoisa) [inline kafka-bridge.go]: It's not necessary to cast httpGtgCheckto StatusChecker.

File: kafka-bridge.go (hunk 3 above)
Comment (davidbalazs93) [inline kafka-bridge.go]: :+1:

File: kafka-bridge.go (hunk 4 above)
Comment (davidbalazs93) [inline kafka-bridge.go]: :+1:

File: circle.yml (hunk 2 above)
Comment (davidbalazs93) [inline circle.yml]: :+1:

File: Dockerfile [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
 RUN apk update \
-  && apk add bash \
   && apk add git bzr \
-  && apk add go libc-dev \
-  && export GOPATH=/gopath \
   && REPO_PATH="github.com/Financial-Times/coco-kafka-bridge" \
   && mkdir -p $GOPATH/src/${REPO_PATH} \
   && mv /kafka-bridge/* $GOPATH/src/${REPO_PATH} \
   && cd $GOPATH/src/${REPO_PATH} \
   && go get -t \
```
Comment (izzyblues) [inline Dockerfile]: If you are using govendor, I think that you do not need this line anymore.

File: Dockerfile [hunk 6]
```diff
@@ -1,18 +1,18 @@
...
-  && go test \
+  && go get -u github.com/kardianos/govendor \
+  && $GOPATH/bin/govendor sync \
+  && $GOPATH/bin/govendor test +local \
```
Comment (izzyblues) [inline Dockerfile]: If you run this test in circle.yml, maybe this line is redundant. Do you agree?

File: circle.yml [hunk 7]
```diff
@@ -1,7 +1,34 @@
...
+  post:
+    # Get the project sources to a proper GO path
+    - mkdir -p $(dirname $GO_PROJECT_PATH)
+    - ln -sf "$HOME/$CIRCLE_PROJECT_REPONAME/" "$GO_PROJECT_PATH"
+dependencies:
+  pre:
+    - go get -u github.com/kardianos/govendor
+  override:
+    - cd $GO_PROJECT_PATH && govendor sync
+    - cd $GO_PROJECT_PATH && go get -t -d -v ./...
```
Comment (izzyblues) [inline circle.yml]: If you specified all your dependencies in `vendor.json`, I think that you do not need this line.

File: healthchecks.go [hunk 8]
```diff
@@ -40,7 +29,7 @@ func (bridge BridgeApp) proxyForwarderHealthcheck() ftHealth.Check {
 		PanicGuide:       "https://sites.google.com/a/ft.com/ft-technology-service-transition/home/run-book-library/kafka-bridge-run-book",
 		Severity:         1,
 		TechnicalSummary: "Forwarding messages is broken. Check if kafka-proxy in coco is reachable.",
-		Checker:          bridge.checkForwardableProxy,
+		Checker:          bridge.producerInstance.ConnectivityCheck,
```
Comment (izzyblues) [inline healthchecks.go]: ❤️

File: healthchecks_test.go [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+	presentTopic = "Concept"
+	nonExistingTopic = "invalid"
+)
+
+type mockTransport struct {
+	responseStatusCode int
+	responseBody       string
+}
+
+type mockProducerInstance struct {
```
Comment (izzyblues) [inline healthchecks_test.go]: I think that the unit tests are more readable and clean if you use the following libraries:
- https://golang.org/pkg/net/http/httptest/ (mock a http server)
- https://godoc.org/github.com/stretchr/testify/mock (mock a generic object)

You can find an example here: https://github.com/Financial-Times/native-ingester/blob/master/resources/healthcheck_test.go

File: vendor/vendor.json [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+	"package": [
+		{
+			"checksumSHA1": "FUhczVSvt/VQAlWfzATo1YLez2E=",
+			"path": "github.com/Financial-Times/go-fthealth/v1a",
+			"revision": "f9074331b3ab48bca0e6d0afc211a673d88ac925",
+			"revisionTime": "2016-06-16T09:04:59Z"
+		},
+		{
+			"checksumSHA1": "yceqWuIbFzntmSHl66OwNoP/c6E=",
+			"path": "github.com/Financial-Times/message-queue-go-producer/producer",
```
Comment (izzyblues) [inline vendor/vendor.json]: In my opinion, it is more convenient if you vendor a specific version of the libraries that you use.
In this way, who is going to make a change to the app knows what's the version of the library in use.
e.g.: `govendor sync github.com/Financial-Times/message-queue-go-producer/producer@v0.1.0`

File: Dockerfile (hunk 5 above)
Comment (davidbalazs93) [inline Dockerfile]: yep, you're right :+1:

File: Dockerfile (hunk 6 above)
Comment (davidbalazs93) [inline Dockerfile]: yes, I will remove this step

File: circle.yml (hunk 7 above)
Comment (davidbalazs93) [inline circle.yml]: you are right :+1:

File: vendor/vendor.json (hunk 10 above)
Comment (davidbalazs93) [inline vendor/vendor.json]: yep, thank you for that, I will add specific versions on libs where possible :+1:

**Pull request merged**